spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java

```diff
@@ -151,6 +157,59 @@ public void testSave() throws Exception {
 	}
 
 	@Test
+	public void testCompose() throws Exception {
+		assertEquals(0, repository.count());
+		TaskDefinition testTask = new TaskDefinition("task", "task");
+		repository.save(testTask);
+		appRegistry.save("task", ApplicationType.task, new URI("http://fake.example.com/"),null);
+		mockMvc.perform(
+				post("/tasks/definitions/compose").param("name", "myTask").param("definition", "task")
+						.accept(MediaType.APPLICATION_JSON)).andDo(print())
+				.andExpect(status().isOk());
+
+		assertEquals(2, repository.count());
+
+		TaskDefinition myTask = repository.findOne("myTask");
+
+		assertEquals(2, myTask.getProperties().size());
+		assertEquals("myTask", myTask.getProperties().get("spring.cloud.task.name"));
+		assertEquals("task", myTask.getProperties().get("graph"));
+		assertEquals("composed-task-runner --graph=\"task\"", myTask.getDslText());
+		assertEquals("myTask", myTask.getName());
+	}
+
+	@Test
+	public void testComposedInvalidParse() throws Exception {
+		assertEquals(0, repository.count());
+		appRegistry.save("task", ApplicationType.task, new URI("http://fake.example.com/"),null);
+		mockMvc.perform(
+				post("/tasks/definitions/compose").param("name", "myTask").
+						param("definition", "task%%$$")
+						.accept(MediaType.APPLICATION_JSON)).andDo(print())
+				.andExpect(status().is5xxServerError())
+				.andExpect(content().json(
```

Comment:
This is too specific of an assertion IMO. Better focus on interesting bits rather than the actual whole